File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def getMinimumDifference(self, root: Optional[TreeNode]) -> int:
        arr = []
        def inorder(root):
            if not root:
                return
            inorder(root.left)
            arr.append(root.val)
            inorder(root.right)
        inorder(root)
        
        ans = float("inf")
        for i in range(len(arr) - 1):
            if abs(arr[i] - arr[i + 1]) < ans:
                ans = abs(arr[i] - arr[i + 1])
        return ans

File: 0530-minimum-absolute-difference-in-bst/0530-minimum-absolute-difference-in-bst.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    List<Integer> nodeValues = new ArrayList<>();
    void dfs(TreeNode root){
        if(root == null){
            return;
        }
        
        dfs(root.left);
        nodeValues.add(root.val);
        dfs(root.right);
        
    }
    public int getMinimumDifference(TreeNode root) {
        dfs(root);
        int ans = Integer.MAX_VALUE;
        for(int i = 1; i < nodeValues.size(); i++){
            if (Math.abs(nodeValues.get(i) - nodeValues.get(i-1)) < ans){
                ans = Math.abs(nodeValues.get(i) - nodeValues.get(i-1));
            }
        }
        return ans;
            
        
    }
}